packages/block-library/src/navigation-link/index.php

```diff
@@ -312,11 +312,45 @@ function build_variation_for_navigation_link( $entity, $kind ) {
 	$title       = '';
 	$description = '';
 
+	// Get default labels based on entity type
+	$default_labels = null;
+	if ( $entity instanceof WP_Post_Type ) {
+		$default_labels = WP_Post_Type::get_default_labels();
+	} elseif ( $entity instanceof WP_Taxonomy ) {
+		$default_labels = WP_Taxonomy::get_default_labels();
+	}
+
+	// Get title and check if it's default
+	$is_default_title = false;
 	if ( property_exists( $entity->labels, 'item_link' ) ) {
 		$title = $entity->labels->item_link;
+		if ( $default_labels && isset( $default_labels['item_link'] ) ) {
```

Comment:
I believe `isset` will already check the existence of `$default_labels` so there's a small (optional) change you could make here:

```suggestion
		if ( isset( $default_labels['item_link'] ) ) {
```